pkg/controller/drain/drain_controller.go

```diff
@@ -381,9 +449,34 @@ func (ctrl *Controller) drainNode(node *corev1.Node, drainer *drain.Helper) erro
 			ctrl.enqueueAfter(node, ctrl.cfg.DrainRequeueDelay)
 		}
 
+		nErr := upgrademonitor.GenerateAndApplyMachineConfigNodes(
+			&upgrademonitor.Condition{State: v1alpha1.MachineConfigNodeUpdateExecuted, Reason: string(v1alpha1.MachineConfigNodeUpdateDrained), Message: fmt.Sprintf("Node Drain has failed")},
```

Comment:
(non-blocking) Just as a random suggestion, I wonder if "Node Drain has not yet succeeded" is a better way to phrase this.

We drain in intervals of 90s (?) so we expect drains to "fail" quite a bit for a node that's running slow-to-terminate workloads. The real failure comes after an hour timeout (there have been asks to customize that and/or force through the drains so our behaviour is not the best atm). Might be good to soften the "error" (or differentiate between the 90s timeout and 1hr timeout)